---
title: "lazappiRNA"
author: "Luke Zappia"
date: "`r Sys.Date()`"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{lazappiRNA}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r global_options, include = FALSE}
# Default RMarkdown options
# Changing these will change them for all chunks unless otherwise set
knitr::opts_chunk$set(
                        autodep        = TRUE,
                        cache          = FALSE,
                        cache.comments = TRUE,
                        collapse       = TRUE,
                        comment        = "##",
                        #dev            = "png",
                        echo           = TRUE,
                        error          = FALSE,
                        fig.width      = 14,
                        fig.height     = 8,
                        highlight      = TRUE,
                        include        = TRUE,
                        message        = FALSE,
                        prompt         = FALSE,
                        results        = "markup",
                        size           = "normalsize",
                        strip.white    = TRUE,
                        tidy           = FALSE,
                        tidy.opts      = NULL,
                        warning        = FALSE
               )
```

Welcome!
========

Welcome to lazappiRNA! Many studies across the life science now make use of
RNA-seq to investigate gene products. These experiments can provide high-quality
measurements of gene expression across a wide dynamic range, and in contrast to
microarracy experiments do not require probes to be designed for specific
regions.

Perhaps the most common analysis pipeline for RNA-seq data is to perform
quality control, align reads to a reference genome, summarise reads against a
known annotation and perform differential expression testing. The aim of this
process is to identify genes or transcripts that have significantly different 
activity levels under some treatment condition (disease, drug, knockout, 
environment...) compared to a control. Further analysis attempts to investigate 
the biological implications of these changes in regulation.

Multiple methods exist for each of these steps. This package takes a matrix of
summarised counts for a group of samples and provides a unified interface for
differential expression testing (using edgeR, DESeq, DESeq2 and limma-voom) and
functional analysis. The aim is allow simultaneous analysis using multiple
packages including robust quality control, easily comparable visulisations and
combination of results.

Before we get started let's load some packages we will need:

```{r libraries}
library(lazappiRNA)

# Plotting
library(ggplot2)

# RNA-seq
library(HTSFilter)
```

Data
====

For this vignette we are going to make use of data from Sultan et al. available
from the _HTSFilter_ package. This dataset contains two biological replicates
from a human embryonic kidney and B cell line. The raw read count and phenotype
table are available as part of the ReCount online resource.

We have alread load _HTSFilter_ so we can start by attaching the data in 
`sultan`. Other information is available but we are just interested in the 
matrix of counts and the groups for each sample:

```{r data}
data("sultan")

counts <- exprs(sultan)
groups <- pData(sultan)$cell.line

rm(sultan)

groups

head(counts)
```

The matrix of counts and factor specifying the group of each sample are going
to be the main input for our analysis.

Exploration
===========

Before starting our analysis we should begin by exploring our data. This allows
us to spot anything that is obviously wrong and provides a baseline we can make
comparisons to later. Let's have a look a the count density.

```{r count-density}
countDensity(counts)
```

All of the plotting function in lazappiRNA return _ggplot2_ objects. These can
be easily modified, for example if we prefer the black and white theme and 
would like a title:

```{r count-density-bw}
countDensity(counts) + theme_bw() + ggtitle("Count Densities")
```

Here we can see that the distribution is both over a large range and heavily
skewed towards low values. For this reason transformed version of counts are
commonly used for plotting, such as: $x = x + 1$

$$ x + 1 $$

$$
\begin{aligned}
    y = log_2(K + 1)
\end{aligned}
$$

More sophisted transformations are provided by the differential expression
packages such as the Variance Stabilising Transformation (_DESeq_),
regularised-log (_DESeq2_) and log Counts Per Million (_edgeR_). lazappiRNA
is capable of performing each of these.

```{r transform}
transformed <- transformCounts(counts,
                               methods = c("log", "vst", "rlog", "logCPM"))

names(transformed)

head(transformed[["log"]])
```

The result of this function is a list of matrices where each is transformed
using one of the specifed methods. We can now use the `listDensity` function
to plot each of these. Again the result is a list but now each item is a 
_ggplot2_ object containing a density plot. An additional item has been added
which holds the combined plot.

```{r densities}
densities <- listDensity(transformed)

names(densities)

densities[["combined"]]
```

Session Info
============

```{r sessionInfo}
sessionInfo()
```

